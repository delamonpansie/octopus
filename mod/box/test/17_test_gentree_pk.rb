#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
rows_per_wal=100

object_space[0].enabled = 1
object_space[0].index[0].type = "TREE"
object_space[0].index[0].unique = 1
object_space[0].index[0].key_field[0].fieldno = 0
object_space[0].index[0].key_field[0].type = "STR"
object_space[0].index[0].key_field[1].fieldno = 1
object_space[0].index[0].key_field[1].type = "STR"
EOD
  end
end

Env.connect_eval do
  ping
  insert ["x16", "y16", 1]
  insert ["x22", "y22", 2]
  select ["x22"], ["x16"]
end
