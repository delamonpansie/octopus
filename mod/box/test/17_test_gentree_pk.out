# box.ping()
:pong

# box.insert(["x16", "y16", 1])
1

# box.insert(["x22", "y22", 2])
1

# box.select(["x22"], ["x16"])
[["x22", "y22", "\x02\x00\x00\x00"], ["x16", "y16", "\x01\x00\x00\x00"]]

